### DISCUSSION TITLE
Maandag 07-10-2024

### DISCUSSION THREAD

**Discussion by Annevd** [rutgerkock/buurtcampus-oost#4] [Stand-ups]
- Hoe gaat het met je?
- Wat wil je deze week bereiken (aan werken)?
- Waar loop je tegen aan?

**Comment by Annevd** [comment]
**Hoe gaat het met je?**

Op dit moment een beetje stressvol. Ik spendeer veel tijd aan het perfectioneren van het menu, met het idee dat het klaar moet zijn om live te gaan over een paar maanden. Dit loopt nog niet heel soepel.

**Wat wil je deze week bereiken (aan werken)?**

Ik heb het idee dat ik misschien wel helemaal opnieuw zal moeten beginnen met het menu om het zonder `:has()` te maken. Nu gaat dat gewoon simpelweg niet door de manier dat de HTML structuur in elkaar zit. Ik hoop dat ik vóór de sprint review nog een mooi bruikbaar menu kan laten zien.

**Waar loop je tegen aan?**

Ik probeer mijn menu te maken met het principe PE. Het menu moet bruikbaar zijn zonder Javascript. Daarvoor gebruik ik nu de `:has()` selector in CSS. Het probleem hiermee is dat dit in een aantal (oudere) browsers niet ondersteund word. 
Het lukt mij niet om dit zonder `:has()` voor elkaar te krijgen. Misschien dat iemand hier nog ideeën voor heeft? Let me know :,)

Ook krijg ik een paar conflicteerde meldingen van W3 HTML validator en Lighthouse, wat bij de één goed is is bij de ander fout.

**Reply by Annevd** [reply]
### Update

Ik heb net met Krijn besproken en voor het `:has()` probleem zijn we tot de conclusie gekomen dat ik dit kan oplossen door `@supports` te gebruiken voor de `:has()` selector, zodat wanneer een browser het niet support het menu er ook niet staat. Zo zijn er geen onbruikbare elementen op de site.

Echter gaf Krijn wel de feedback dat dit menu misschien niet zo hard nodig is. Wil je dit menu écht werkend hebben voor iedereen, dan moet je veel tijd en moeite stoppen om het volledig via server side rendering te doen zodat de keuzes worden opgeslagen en dat dark mode ook ingesteld wordt aan de hand van je OS. 

Zijn feedback was dat dit eigenlijk teveel werk is voor wat je krijgt, mensen die deze opties namelijk nodig hebben weten wél dat het al in de browser ingebouwd zit, en dus is dit niet per se nodig in de website zelf.

**Reply by Annevd** [reply]
### Update 2

Na nog een keer sparren met Koop en Krijn zijn we toch tot de conclusie gekomen het menu wél te houden. Omdat je anders een bepaalde doelgroep buitensluit (de mensen die niet weten dat deze functies al in de browser ingebouwd zijn). In mijn geval gebruik je dan gewoon de `@supports` zodat het niet zichtbaar is wanneer een browser het niet support.

Het opslaan van de voorkeuren is niet per se nodig, omdat mensen deze website niet super vaak zullen bezoeken.

Wél is het een idee dat de website start in het kleurenschema (licht of donker) van jouw OS systeem. Deze kan je dan overrulen wanneer je toch een ander kleurenschema wil. Dit zou kunnen aan de hand van een `media query` in CSS. Hier is in principe geen Javascript voor nodig.

**Comment by rutgerkock** [comment]
**Hoe gaat het met je?**
Goed, bezig met Harry nu het weer af was. Snap niet veel van hoe Harry kwa svg's/css in elkaar zit. Maar Larissa had dat aardig uitgezocht dus konden we dat mooi aan het weer koppelen. Dat is gelukt, ben ik blij mee.

**Wat wil je deze week bereiken (aan werken)?**
Harry implementeren in de website op een speelse manier. Harry is nu nog nergens te vinden, ik hoop dat het lukt om ervoor te zorgen dat Harry op een slimme manier weergegeven wordt met nuttige informatie op verschillende pagina's. Plan is om dit de komende dagen te regelen, of in ieder geval de basis op te zetten zodat we hem daarna makkelijk op verschillende pagina's met verschillende teksten kan toevoegen.

**Waar loop je tegen aan?**
De website is niet door ons opgezet en een beetje een doolhof. Het is soms ingewikkeld CSS/HTML/Svelte structuur te begrijpen, de inspector helpt hier eg mee. Heb er soms moeite mee om uit te vogelen waar ik een element moet maken, in welk van de 1294 mapjes, maar kom hier met een beetje assistentie wel prima uit.

Ook de logica rondom Harry is even nadenken met if-statements en waardes die veranderen en wat dan een logische manier is om iets te bedenken, beetje vermoeiend maar wel te doen.

**Comment by Lmikkers** [comment]
**Hoe gaat het met je?**
Goed! In de ochtend zijn Rutger en ik bezig geweest met Harry. Rutger was hiervoor bezig met de weer API en ik met Harry in de code toevoegen. We zijn nu bezig om dit met elkaar te koppellen, dit is vanochtend gelukt!

**Wat wil je deze week bereiken (aan werken)**
Dat Harry op een speelse manier op de website komt, op elke pagina (home, stekjes en zaden) en dat de data dynamisch wordt uitgeladen. Rutger en ik gaan hier samen verder aan werken en willen dit koppelen aan Hygraph, we moeten waarschijnlijk wel een nieuw veld toevoegen. 

**Waar loop je tegen aan**
Dat ik nog niet met Hygraph heb gewerkt en dat ik geen veld kan toevoegen, maar dit hebben we aan Joost gevraagd. Omdat we in iemands anders project werken is het soms lastig dingen te vinden, normaal doe ik dit in de inspector, maar nu met Svelte kit is dat wat lastiger, door de classes die Svelte aan maakt. Ik kom hier wel uit, maar het duurt wat langer dan normaal hierdoor. 

We hebben Harry nu toegevoegd en aan het weer gekoppeld en hebben hiervoor een extra css aangemaakt. Maar we moeten even kijken of we dit op een nette manier doen en ook `DRY` doen.

**Comment by Nazneen05x** [comment]
- **Hoe gaat het met je?**

Ik ben helaas ziek, dus het kan veel beter. Afgelopen weken ben ik bezig geweest met de agenda-carousel. De agenda-carousel is voor een groot deel af, nu puntjes op de I (responsief enz)  zetten.

- **Wat wil je deze week bereiken (aan werken)?**

Deze week wil ik even focussen op een andere taak, de geveltuin pagina wil ik deze week gaan oppakken. 

**Issues:**

[#37](https://github.com/users/Nazneen05x/projects/14/views/1?pane=issue&itemId=82435059&issue=Nazneen05x%7CS4-S19-Bieb-In-Bloei%7C37)

[#38](https://github.com/users/Nazneen05x/projects/14/views/1?pane=issue&itemId=82435530&issue=Nazneen05x%7CS4-S19-Bieb-In-Bloei%7C38)

[#39](https://github.com/users/Nazneen05x/projects/14/views/1?pane=issue&itemId=82435781&issue=Nazneen05x%7CS4-S19-Bieb-In-Bloei%7C39)

[#40](https://github.com/users/Nazneen05x/projects/14/views/1?pane=issue&itemId=82436213&issue=Nazneen05x%7CS4-S19-Bieb-In-Bloei%7C40)

[#41](https://github.com/users/Nazneen05x/projects/14/views/1?pane=issue&itemId=82436410&issue=Nazneen05x%7CS4-S19-Bieb-In-Bloei%7C41)

- Waar loop je tegen aan?
Ik moet nog beginnen dus op dit moment nog niks.
